### PR TITLE
Build and run with Java 11

### PR DESCRIPTION
**Pull request by bpetermannS11**
Build and run Midonet with Java 11

Migrate Midonet to Java 11.
The project should be built with JDK 11 and Midonet runtime now
requires Java 11.
The template env scripts now confire Shenandoah GC

Updated gradle to 6.8.3 and updated gradle plugins.
Adapted build configuration to be in line with the new gradle.

Include jaxb runtime explicitly in dependencies because it's not part
of the standard openjdk releases anymore.

Updated some dependencies to their latest patch release (not upgrading
to a higher major release though): akka, jetty, jsch, netty,
zookeeper. Use latest scala 2.11 (2.11.12).

Use maven repo https URLs instead of http.

Fix compilation of scala code and fix some deprications in scala and
java code.

Let MutableUUID behave like java UUID: trailing dashes in a UUID
string lead are not accepted anymore. Adapt MutableUUID to be in line.

### CODE REVIEW

File: midonet-tools/src/share/prepare-java [hunk 1]
```diff
@@ -17,19 +17,14 @@
...
-                 /usr/$LIB_PATTERNS/jvm/java-8-oracle \
-                 /usr/$LIB_PATTERNS/jvm/zulu-8-amd64 \
-                 /usr/$LIB_PATTERNS/jvm/jre-1.8.0-openjdk \
-                 /usr/$LIB_PATTERNS/jvm/jre-8-oracle-x64 \
-                 /usr/$LIB_PATTERNS/jvm/jdk-8-oracle-x64)
+                 /usr/$LIB_PATTERNS/jvm/adoptopenjdk-11-hotspot-amd64 \
+                 )
 
 # zulu prints "openjdk version" while openjdk and oracle print "java version"
 check_for_java8() {
```
Comment (stefreak) [inline midonet-tools/src/share/prepare-java]: is this now actually checking for java 11?

Comment (bpetermannS11) [reply midonet-tools/src/share/prepare-java]: Fixed

File: gradlew [hunk 2]
```diff
@@ -228,11 +170,19 @@ if $cygwin ; then
     esac
 fi
 
-# Split up the JVM_OPTS And GRADLE_OPTS values into an array, following the shell quoting and substitution rules
-function splitJvmOpts() {
-    JVM_OPTS=("$@")
+# Escape application args
+save () {
```
Comment (stefreak) [inline gradlew:174]: Did you mean `safe`?

**Review by stefreak — APPROVED** [review]
LGTM :shipit:  🚢

File: build.gradle [hunk 3]
```diff
@@ -201,6 +204,10 @@ allprojects {
         dependencies { compile libraries.scala }
 
         tasks.withType(ScalaCompile) {
+            scalaCompileOptions.forkOptions.with {
+                memoryMaximumSize = '1g'
+		// jvmArgs = ['--add-exports', 'java.base/sun.nio.ch=ALL-UNNAMED']
```
Comment (stefreak) [inline build.gradle]: why is this commented out?

File: build.gradle [hunk 4]
```diff
@@ -141,7 +144,7 @@ project.ext {
         slf4j: "org.slf4j:slf4j-api:1.7.7",
         snappy: "org.xerial.snappy:snappy-java:1.1.2.4",
         sqlite: "org.xerial:sqlite-jdbc:3.8.7",
-        sun_tools: project.files("${System.properties['java.home']}/../lib/tools.jar"),
+        //sun_tools: project.files("${System.properties['java.home']}/../lib/tools.jar"),
```
Comment (stefreak) [inline build.gradle]: why is this commented out?

File: gradlew (hunk 2 above)
Comment (bpetermannS11) [reply gradlew:174]: gradlew is generated by gradle and I didn't edit it

File: build.gradle (hunk 3 above)
Comment (bpetermannS11) [reply build.gradle]: not needed here - will remove it

File: build.gradle (hunk 4 above)
Comment (bpetermannS11) [reply build.gradle]: tools.jar doesn't exist anymore and it doesn't seem to be needed - I will delete the line

**Pull request merged**